packages/zen-bull-netting/src/NettingLib.sol

```diff
@@ -0,0 +1,180 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+// interface
+import { IERC20 } from "openzeppelin/interfaces/IERC20.sol";
+import { IZenBullStrategy } from "./interface/IZenBullStrategy.sol";
+import { IOracle } from "./interface/IOracle.sol";
+import { IEulerSimpleLens } from "./interface/IEulerSimpleLens.sol";
+import { IWETH } from "./interface/IWETH.sol";
+
+library NettingLib {
+    event DepositAuction(address indexed trader, uint256 indexed bidId, uint256 quantity);
+    event WithdrawAuction(
+        address indexed trader, uint256 indexed bidId, uint256 quantity, uint256 price
+    );
+
+    function transferWethFromMarketMakers(
+        address weth,
+        address trader,
+        uint256 quantity,
+        uint256 oSqthToMint,
+        uint256 clearingPrice
+    ) external returns (bool, uint256) {
+        if (quantity >= oSqthToMint) {
+            IWETH(weth).transferFrom(trader, address(this), (oSqthToMint * clearingPrice) / 1e18);
+
+            return (true, 0);
+        } else {
+            IWETH(weth).transferFrom(trader, address(this), (quantity * clearingPrice) / 1e18);
+
+            return (false, (oSqthToMint - quantity));
+        }
+    }
+
+    function transferOsqthFromMarketMakers(
+        address oSqth,
+        address trader,
+        uint256 toExchange,
+        uint256 quantity
+    ) internal returns (bool, uint256) {
+        if (quantity < toExchange) {
+            IERC20(oSqth).transferFrom(trader, address(this), quantity);
+            return (false, (toExchange - quantity));
+        } else {
+            IERC20(oSqth).transferFrom(trader, address(this), toExchange);
+            return (true, 0);
+        }
+    }
+
+    function transferOsqthToMarketMakers(
+        address oSqth,
+        address trader,
+        uint256 bidId,
+        uint256 oSqthBalance,
+        uint256 quantity
+    ) external returns (bool, uint256) {
+        if (quantity < oSqthBalance) {
+            IERC20(oSqth).transfer(trader, quantity);
+
+            emit DepositAuction(trader, bidId, quantity);
+
+            return (false, (oSqthBalance - quantity));
+        } else {
+            IERC20(oSqth).transfer(trader, oSqthBalance);
+
+            emit DepositAuction(trader, bidId, oSqthBalance);
+
+            return (true, 0);
+        }
+    }
+
+    function transferWethToMarketMaker(
```

Comment:
why did you do this one differently from the other weth one?

Reply:
which one ?